.github/workflows/release.yml

```diff
@@ -71,18 +41,40 @@ jobs:
       - name: Install dependencies
         run: |
           apt-get update
-          apt-get install -y build-essential cmake gettext locales ninja-build pkg-config unzip
+          apt-get install -y build-essential cmake gettext ninja-build unzip
       - if: github.event_name == 'push' || (github.event_name == 'workflow_dispatch' && github.event.inputs.tag_name != 'nightly')
         run: CC=gcc-10 make appimage-latest
       - if: github.event_name == 'schedule' || (github.event_name == 'workflow_dispatch' && github.event.inputs.tag_name == 'nightly')
         run: CC=gcc-10 make appimage-nightly
+      - if: github.event_name == 'push' || (github.event_name == 'workflow_dispatch' && github.event.inputs.tag_name != 'nightly')
+        run: printf 'NVIM_BUILD_TYPE=Release\n' >> $GITHUB_ENV
+      - if: github.event_name == 'schedule' || (github.event_name == 'workflow_dispatch' && github.event.inputs.tag_name == 'nightly')
+        run: printf 'NVIM_BUILD_TYPE=RelWithDebInfo\n' >> $GITHUB_ENV
       - uses: actions/upload-artifact@v3
         with:
           name: appimage
           path: |
             build/bin/nvim.appimage
             build/bin/nvim.appimage.zsync
           retention-days: 1
+      - name: Build release
+        id: build
+        run: |
+          make distclean
```

Comment:
do we need to distclean here? does `make appimage` somehow invalidate the build? ideally we would just produce the tar.gz artifact from the same build of nvim as the appimage.

Reply:
Unsure. The appimage code is very difficult to read and unorganized. I didn't want to take a risk. I can investigate.

Reply:
and if you flip it? build normally first, package the tar, then run the appimage workflow (without `make distclean`)?

Reply:
Nothing works. Packaging tar and the appimage fails. Packaging appimage and then tar will make the Release type not work, as the appimage is always built with RelWithDebInfo. The only realistic way I can resolve this without `make distclean` is to eliminate or heavily refactor `script/genappimage.sh` to try to make sense whatever is written there.

Reply:
So the root issue is that the appimage and the tar.gz was built using different build types? Seems we have to pick one for each build type (nightly and release) and use that for both artifacts.

As it is the recommended build I'd assume the choices for appimage is fine, so we could just build the appimage first and then `cpack` whatever the appimage script already is producing.

Reply:
Yes, that works.